GetStream/stream-chat-android
# [2902] Eliminate `ChatDomain::deleteChannel`

Closes #2902 

### 🎯 Goal

In order to merge `ChatDomain` into LLC, we need to move all code from ChatDomain into LLC and `OfflinePlugin`.

### 🛠 Implementation details
- A new `ChatClient::deleteChannel` extension function is added.
- `DeleteChannel` usecase is removed.
- `ChatDomain::deleteChannel` is deprecated.

### 🧪 Testing

1. Run any sample app.
2. Try to Delete any channel for which you have permission to delete.
2. Make sure the channel is deleted as expected. 

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
